src/main/java/compal/model/tasks/Task.java

```diff
@@ -0,0 +1,395 @@
+package compal.model.tasks;
+
+import java.io.Serializable;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Represents task with description, status and reminder.
+ */
+public abstract class Task implements Serializable {
+
+    public boolean isDone;
+
+    //***Class Properties/Variables***--------------------------------------------------------------------------------->
+    protected String symbol;
+    private String description;
+    private Date date;   //For now, we only process dates in the format dd/mm/yyyy hhmm. See TaskList class for details
+    private Date startTime;
+    private Date endTime;
+    private boolean hasReminder;
+    private Priority priority;
+    private long priorityScore;
+    private int id;
+
+
+    /**
+     * Constructs Task object.
+     *
+     * @param description Description.
+     */
+    protected Task(String description, Priority priority) {
+        this.description = description;
+        this.priority = priority;
+        this.isDone = false;
+        hasReminder = false;
+
+    }
+
+
+    /**
+     * Gets priority status (HIGH, MEDIUM, LOW) of task.
+     *
+     * @return Priority status of task.
+     */
+    public Priority getPriority() {
+        return priority;
+    }
+
+    /**
+     * Gets/returns task id.
+     *
+     */
+    public int getId() {
+        return id;
+    }
+
+
+
+
+
+    /**
+     * Sets priority of task as HIGH, MEDIUM or LOW.
+     *
+     * @param priority Priority of task.
+     */
+    public void setPriority(Priority priority) {
+        this.priority = priority;
+    }
+
+
+    /**
+     * Sets task id.
+     *
+     */
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    /**
+     * Gets status icon (tick or cross) of task.
+     *
+     * @return Status icon (tick or cross) of task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718");
```

Comment:
"Magic strings" should be avoided. Especially so when you have put in so much effort to avoid them but make the mistake ehre.